SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -448,40 +448,28 @@
 
   <Target Name="SetRoslynCodeAnalysisProperties">
 
-    <PropertyGroup>
-      <!-- Check if the ruleset exists -->
-      <SonarQubeRoslynRulesetFullName>$(SonarQubeConfigPath)SonarQubeRoslyn-cs.ruleset</SonarQubeRoslynRulesetFullName>
-      <SonarQubeRoslynRulesetExists>$([System.IO.File]::Exists($(SonarQubeRoslynRulesetFullName)))</SonarQubeRoslynRulesetExists>
-
-      <!-- Check if the downloaded analyzer assemblies exists. We assume that one of the assemblies is called "SonarLint.dll" for the
-           purposes of locating the analyzers. -->
-      <SonarLintAnalyzerDir>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), SonarLint.dll))</SonarLintAnalyzerDir>
-      <SonarLintAnalyzerDir Condition=" $(SonarLintAnalyzerDir) != '' AND !HasTrailingSlash('$(SonarLintAnalyzerDir)')">$(SonarLintAnalyzerDir)\</SonarLintAnalyzerDir>
-      <SonarLintAnalyzerAssembly Condition=" $(SonarLintAnalyzerDir) != '' ">$(SonarLintAnalyzerDir)SonarLint.dll</SonarLintAnalyzerAssembly>
-      <SonarQubeRoslynAssemblyExists>$([System.IO.File]::Exists($(SonarLintAnalyzerAssembly)))</SonarQubeRoslynAssemblyExists>
-
-      <SonarLintFound Condition="
-          $(SonarQubeRoslynRulesetExists) == 'true'
-          AND $(SonarQubeRoslynAssemblyExists) == 'true'
-          AND $(Language) == 'C#'
-          ">true</SonarLintFound>
-      <SonarLintFound Condition=" $(SonarLintFound) == '' ">false</SonarLintFound>
-    </PropertyGroup>
+    <!-- Fetch the relevant settings from the config file -->
+    <GetAnalyzerSettings AnalysisConfigDir="$(SonarQubeConfigPath)">
+      <Output TaskParameter="RuleSetFilePath" PropertyName="SQRuleSetFilePath" />
+      <Output TaskParameter="AnalyzerFilePaths" ItemName="SQAnalyzerFilePaths" />
+      <Output TaskParameter="AdditionalFiles" ItemName="SQAdditionalFiles" />
+    </GetAnalyzerSettings>
 
-    <PropertyGroup Condition=" $(SonarLintFound) == 'true' ">
+    <PropertyGroup>
       <ErrorLog Condition=" $(ErrorLog) == '' " >$(TargetDir)SonarQube.Roslyn.ErrorLog.json</ErrorLog>
-      <ResolvedCodeAnalysisRuleSet>$(SonarQubeRoslynRulesetFullName)</ResolvedCodeAnalysisRuleSet>
+      <ResolvedCodeAnalysisRuleSet>$(SQRuleSetFilePath)</ResolvedCodeAnalysisRuleSet>
     </PropertyGroup>
 
-    <ItemGroup Condition=" $(SonarLintFound) == 'true' ">
+    <ItemGroup>
       <!-- Remove any existing analyzers and additional files -->
       <Analyzer Remove="@(Analyzer)" />
       <AdditionalFiles Remove="@(AdditionalFiles)" />
 
-      <!-- Add SonarLint analyzers and additional files -->
-      <Analyzer Include="$(SonarLintAnalyzerDir)*SonarLint*.dll" />
-      <AdditionalFiles Include="$(SonarQubeConfigPath)SonarLint.xml" />
+      <!-- Add specified analyzers and additional files (if any) -->
```

Comment:
There might not be any Roslyn analyzers or additional files, which is fine (means the profile didn't contain any).